src/main/java/com/vinylteam/vinyl/service/impl/VinylShopService.java

```diff
@@ -0,0 +1,104 @@
+package com.vinylteam.vinyl.service.impl;
+
+import com.vinylteam.vinyl.entity.Vinyl;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import com.vinylteam.vinyl.service.ShopService;
+
+import java.io.IOException;
+import java.util.*;
+
+public class VinylShopService implements ShopService {
+    private final String startLink = "http://vinyl.ua";
+
+    HashSet<String> getShowcaseLinks() throws IOException {
+        HashSet<String> showcaseLinks = new HashSet<>();
+        Document doc = Jsoup.connect(startLink).get();
+        Elements innerLinks;
+        innerLinks = doc.getElementsByTag("a");
+
+        for (Element innerLink : innerLinks) {
+            String link;
+            if (innerLink.attr("href").contains("www") || innerLink.attr("href").contains("http")) {
+                link = innerLink.attr("href");
+            } else {
+                link = startLink + innerLink.attr("href");
+            }
+            if (link.contains("http://vinyl.ua/showcase/")) {
+                showcaseLinks.add(link);
+            }
+        }
+
+        return showcaseLinks;
+    }
+
+    HashSet<String> getPageLinks(HashSet<String> showcaseLinks) throws IOException {
```

Comment:
How do you avoid getting repeating showcase links here?
Like when you go through all links you also collect "electronic, classic, rock..." again, right?
How do you avoid collecting them?